requirements.txt

```diff
@@ -7,3 +7,7 @@ jinja2
 PyYAML
 cryptography
 packaging
+# NOTE: resolvelib 0.x version bumps should be considered major/breaking
+# NOTE: and we should update the upper cap with care, at least until 1.0
+# NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
+resolvelib >= 0.5.3, < 0.6.0  # dependency resolver used by ansible-galaxy
```

Comment:
So do we run the risk here that pip has it's own pinned resolvelib version but we have another or does pip vendor it entirely and we don't care about it?

Reply:
It's one of the libs pip re-vendors every once in a while: https://github.com/pypa/pip/tree/master/src/pip/_vendor#vendoring-policy. For example: https://github.com/pypa/pip/pull/9369.

Reply:
So I'd say we don't care. We'll need to upgrade the upper bound once in a while along with updating our code to work with newer public API when necessary.

Reply:
Not entirely correct.... Linux distros typically unvendor libraries.  So you could still end up with a problem if you pin your versions too strictly.

Reply:
@abadger in this case we know for sure that v0.6.0 will add extra arguments w/o defaults to some hooks 100% breaking the API so before changing the upper bound, we'll have to adjust that. Besides, we now have a designated location in the tree where the packagers can drop-in libs w/o having to package them separately if they wish so.

But I see your point: deb packages infamous for regularly breaking pip and other Python packaging-related things. Ref: https://gist.github.com/tiran/2dec9e03c6f901814f6d1e8dad09528e.

Anyway... We have to specify what exactly is supported right now, this info gets into the metadata and the packagers are responsible for not breaking stuff. If we say that 0.6.0 is not supported and they ignore the instruction, it's not our fault, is it?